woocommerce/google-listings-and-ads
# Tweak Budget recommendations logic

### Changes proposed in this Pull Request:

This PR changes the logic in budget recommendations.

- Adapt Budget Recommendation Controller for using `daily_budget` (single column) instead of a range
- Adapt Budget Recommendation Frontend for using `daily_budget` (single column) instead of a range
- Migrate the database tables
- Adapt tests

⚠️  The Budget Recommendation File is not final. We are waiting for a final version with all the currencies. YOu an use the one provided as a sample.

⚠️  The tracking requisite is not implemented in this PR

Possible setup for testing:

- All countries - USD
- Spain and France - Euro


### Screenshots:

<img width="747" alt="Screenshot 2023-11-13 at 11 26 48" src="https://github.com/woocommerce/google-listings-and-ads/assets/5908855/7cbc9d7d-347e-451f-ad9d-27653307b1e4">


### Detailed test instructions:


1. Change the Applicable version to `2.5.11` in `src/DB/Migration/Migration20231109T1653383133.php`
2. Go to the Database and change `gla_file_version` and `gla_db_version` to `2.5.10` in `wp_options` table to refresh the data and migrate the columns. 
3. Re-activate the plugin.
4. Verify that the sample Budget Recommendations has been loaded in `wp_gla_budget_recommendations ` and contains. the right values and columns 
5. Go to the Onboarding, select Spain and France as the Audience, and Euro as the currency
6. COmplete the setup and create an Ads Campaign
7. In the budget notice see the recommended budget as the highest of both countries in EUR 
8. Introduce a budget less than this value and see the message "With a lower budget than your competitor..."
9. Change the budget to equal or more than the suggested one
10. See the message mentioned in step 8 disappear.


### Additional details:

PT --> pcTzPl-1Uo-p2



### Changelog entry

> Tweak - Use a single daily budget instead of a range
